pkgs/development/tools/build-managers/apache-maven/default.nix

```diff
@@ -1,21 +1,43 @@
-{ lib, stdenv, fetchurl, jdk, makeWrapper }:
+{ lib
+, stdenvNoCC
+, fetchurl
+, jdk
+, makeWrapper
+, callPackage
+}:
 
 assert jdk != null;
 
-stdenv.mkDerivation rec {
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "apache-maven";
   version = "3.9.2";
 
-  builder = ./builder.sh;
-
   src = fetchurl {
-    url = "mirror://apache/maven/maven-3/${version}/binaries/${pname}-${version}-bin.tar.gz";
-    sha256 = "sha256-gJ7zIgxtF5GVwGwyTLmm002Oy6Vmxc/Y64MWe8A0EX0=";
+    url = "mirror://apache/maven/maven-3/${finalAttrs.version}/binaries/${finalAttrs.pname}-${finalAttrs.version}-bin.tar.gz";
+    hash = "sha256-gJ7zIgxtF5GVwGwyTLmm002Oy6Vmxc/Y64MWe8A0EX0=";
```

Comment:
I don't think it makes sense to allow `pname` and `version` to be overridable here since the hash isn't.

Reply:
The idea is to have `finalAttrs.finalPackage`, which would not be easily accessible if `rec` is still kept. One might argue whether `lib.fix finalPackage (stdenv.mkDerivation rec { ... })` would do the job, but it doesn't seem very elegant to me.

Reply:
Thank you for explaining. That part makes sense to me. I mean that if someone does override `pname` and `version`, then this source will be broken since the hash will likely be different, so I feel these should not be overridden here.

Reply:
I actually expect a user to override `pname` or `version` at the same time as `src`, whether `rec` is present or not.

Reply:
Yeah, that's fair. It would be broken before too.